gemini/sample-apps/finance-advisor-spanner/pages/2_Semantic_Search.py

```diff
@@ -0,0 +1,67 @@
+"""This module is the page for Semantic Search feature"""
+
+# pylint: disable=line-too-long, invalid-name, import-error, use-dict-literal, duplicate-code
+
+from css import favicon, footer
+from database import display_spanner_query, semantic_query, semantic_query_ann
+from itables.streamlit import interactive_table
+import streamlit as st
+
+st.set_page_config(
+    layout="wide",
+    page_title="FinVest Advisor",
+    page_icon=favicon,
+    initial_sidebar_state="expanded",
+)
+st.logo(
+    "https://storage.googleapis.com/github-repo/generative-ai/sample-apps/finance-advisor-spanner/images/investments.png"
+)
+
+
+def asset_semantic_search() -> None:
+    """This function implements Semantic Search feature"""
+
+    st.header("FinVest Fund Advisor")
+    st.subheader("Semantic Search")
+
+    st.columns([0.25, 0.25, 0.20, 0.10])
+    classes = ["display", "compact", "cell-border", "stripe"]
+    buttons = ["pageLength", "csvHtml5", "excelHtml5", "colvis"]
+    style = "table-layout:auto;width:auto;margin:auto;caption-side:bottom"
+    it_args = dict(
+        classes=classes,
+        style=style,
+    )
+
+    if buttons:
+        it_args["buttons"] = buttons
+
+    query_params = []
+    query_params = [investment_strategy.strip(), investment_manager.strip()]
+
+    with st.spinner("Querying Spanner..."):
+        if annVsKNN == "KNN":
+            return_vals = semantic_query(query_params)
+        else:
```

Comment:
More duplication from other files. Please try to consolidate.

Reply:
fixed. moved common code to Home.py